server/src/main/java/io/druid/segment/realtime/plumber/RealtimePlumber.java

```diff
@@ -187,12 +187,19 @@ public void startJob()
   @Override
   public int add(InputRow row) throws IndexSizeExceededException
   {
-    final Sink sink = getSink(row.getTimestampFromEpoch());
-    if (sink == null) {
+    try {
+      final Sink sink = getSink(row.getTimestampFromEpoch());
+      if (sink == null) {
+        return -1;
+      }
+
+      return sink.add(row);
+    } catch (Exception e) {
+      log.makeAlert(e, "Failed to add row[%s]", schema.getDataSource())
```

Comment:
This can potentially emit a ton of alerts. Can we instead emit the alert once and track how many rows are experiencing problems?